lib/User/User.php

```diff
@@ -460,23 +464,51 @@ public function updateQuota($valueFromLDAP = null) {
 		if($this->wasRefreshed('quota')) {
 			return;
 		}
-		//can be null
-		$quotaDefault = $this->connection->ldapQuotaDefault;
-		$quota = $quotaDefault !== '' ? $quotaDefault : null;
-		$quota = !is_null($valueFromLDAP) ? $valueFromLDAP : $quota;
 
+		$quota = false;
 		if(is_null($valueFromLDAP)) {
 			$quotaAttribute = $this->connection->ldapQuotaAttribute;
 			if ($quotaAttribute !== '') {
 				$aQuota = $this->access->readAttribute($this->dn, $quotaAttribute);
 				if($aQuota && (count($aQuota) > 0)) {
-					$quota = $aQuota[0];
+					if ($this->verifyQuotaValue($aQuota[0])) {
+						$quota = $aQuota[0];
```

Comment:
I'm confused here even when trying to read the above code.

Can you add PHPDoc to remove all doubts ?

Is `$valueFromLDAP` the value of quota attribute from the specific user read from LDAP ?
If yes, then what is the difference between `$valueFromLDAP` and the `ldapQuotaAttribute` value that is read at this line ?